hocs/withRouteGaurd/index.tsx

```diff
@@ -0,0 +1,35 @@
+import type { NextPage } from "next";
+import { useRouter } from "next/router";
+import { MemberInfo } from "~/contexts";
+
+const withRouteGuard = <P extends object>(
+  option: "prevented" | "private",
+  Page: NextPage<P>
+) => {
```

Comment:
`prevented`, `private` 모두 카카오 로그인이 된 상태가 기준이라는 점이 설명을 들어야 이해가 될 것 같아요.
아래 같이 의미를 담아서 변경하는 건 어떨까요? 백엔드에서 사용하는 용어(`CONNECTED`, `ANNONYMOUS` 등)와는 독립적으로 가져갔습니다.

생각이 안나서 Django Permission 클래스대로 생각하긴 했는데, 😂 더 좋은 이름이 있으면 그걸로 해두 될 것 같아요.

```suggestion
const withRouteGuard = <P extends object>(
  permission: "onlyAuthenticated" | "onlyNotAuthenticated" | "allowAny",
  Page: NextPage<P>
) => {
```

Reply:
permission 좋은 거 같네요! 누구의 기준에서 열린 route인건지 명확하게 좋은 것 같습니다!

Reply:
아래 같은 방식으로 각 페이지에서 더 늘어날 수 있는 accountConnectType에 따라 수정할 수 있게 변경했어요
이렇게 괜찮지 않을까요??

[3cb22d7](https://github.com/mash-up-kr/Ladder_Web/pull/27/commits/3cb22d728e311b193e829b9800094468c82d909a)